### PR TITLE
Change the DisableCentralPackageVersions to EnableCentralPackageVersions

### PR DESCRIPTION
**Pull request by cristinamanum**
Change the DisableCentralPackageVersions to EnableCentralPackageVersions based on feedback.

### CODE REVIEW

**Review by zivkan — COMMENTED** [review]
👍 positive properties are much better than negative, to avoid double negatives when the value is false.

**Review by zivkan — APPROVED** [review]
oops, meant to approve with my previous review.

**Review by dtivel — APPROVED** [review]

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets [hunk 1]
```diff
@@ -58,8 +58,8 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Mark that this targets file supports package download. -->
     <PackageDownloadSupported>true</PackageDownloadSupported>
     <!-- Flag if the Central package file is enabled -->
-    <DisableCentralPackageVersions Condition="'$(DisableCentralPackageVersions)' == ''">true</DisableCentralPackageVersions>
-    <_CentralPackageVersionsEnabled Condition="'$(DisableCentralPackageVersions)' != 'true' AND '$(CentralPackageVersionsFileImported)' == 'true'">true</_CentralPackageVersionsEnabled>
+    <EnableCentralPackageVersions Condition="'$(EnableCentralPackageVersions)' == ''">false</EnableCentralPackageVersions>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets:61]: You don't need this line.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets:61]: Is this opt-in temporary? 

See: https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions#opt-in-central-package-version-management

Comment (cristinamanum) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets:61]: This flag will be disabled until the entire feature is ready.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets:61]: You don't need to set anything to false, empty is fine.

In MSBuild it's a bad pattern to set a default that no one respects. 

Specifically just checking for the 
> `EnableCentralPackageVersions` == 'true' 

is enough.

**Comment by AArnott** [comment]
@cristinamanum  This regressed building the VS repo. The Microsoft.Build.CentralPackageVersions already defines and uses this `EnableCentralPackageVersions` property. It sets it to *false* if the packages.props file doesn't exist, but otherwise considers the feature enabled. 

https://github.com/microsoft/MSBuildSdks/blob/f4e31079565777df7656a358777f8db285c4c0bd/src/CentralPackageVersions/Sdk/Sdk.targets#L28-L29

But _this_ PR takes an empty property and sets it to `false` instead, which breaks everyone who used the CPV SDK already since the SDK will respect an explicit `false` value.

Why is NuGet setting this property anyway? Is it intentionally overloading the property used by the CPV SDK? Shouldn't the default behaviors of these two not conflict with each other?

CC: @rainersigwald @MarkOsborneMS

**Comment by AArnott** [comment]
Another problem that I observe is that changing the `EnableCentralPackageVersions` property as defined by an imported file does *not* force a fresh nuget restore. I can change the property and run restore, but nuget will skip restore because relevant inputs haven't changed. But surely this is a very significant input that *has* changed, and should be included in NuGet's up-to-date check.

**Comment by cristinamanum** [comment]
> @cristinamanum This regressed building the VS repo. The Microsoft.Build.CentralPackageVersions already defines and uses this `EnableCentralPackageVersions` property. It sets it to _false_ if the packages.props file doesn't exist, but otherwise considers the feature enabled.
> 
> https://github.com/microsoft/MSBuildSdks/blob/f4e31079565777df7656a358777f8db285c4c0bd/src/CentralPackageVersions/Sdk/Sdk.targets#L28-L29
> 
> But _this_ PR takes an empty property and sets it to `false` instead, which breaks everyone who used the CPV SDK already since the SDK will respect an explicit `false` value.
> 
> Why is NuGet setting this property anyway? Is it intentionally overloading the property used by the CPV SDK? Shouldn't the default behaviors of these two not conflict with each other?
> 
> CC: @rainersigwald @MarkOsborneMS

@AArnott , I did not realized the naming conflict. The two should not conflict with each other. Add an issue on nuget side https://github.com/NuGet/Home/issues/9235

**Pull request merged**